### PR TITLE
fix(engine sequence): improve BPM execution sequence

### PR DESCRIPTION
**Pull request by educhastenier**
Closes [BS-16222](https://bonitasoft.atlassian.net/browse/BS-16222)

### CODE REVIEW

File: md/execution-sequence-states-and-transactions.md [hunk 1]
```diff
@@ -4,127 +4,137 @@ This page describes the sequence of actions performed when a process instance is
...
-11. Completing the instantiation The Engine instantiates and executes the "on enter" connectors (evaluate input expression, execute, evaluate output operation). This an asynchronous operation. Execution duration of a connector is limited to a maximum of 5 minutes by default. 
-12. The process is instantiated, the API call finish and the Engine executes the process flow asynchronously.
+7. The Engine creates and initializes business data
+1. **Subscription only**: The Engine creates and initializes search indexes
+1. The Engine initializes process instance Documents with default value coming from the definition or with submitted files
+1. Operations (that does not initialize data) are executed. Operations at process instantiation level can either come from V6 forms or from calling process instances (for processes stared by call-activities)
+1. The Engine instantiates and executes the "on enter" connectors (evaluate input expression, execute, evaluate output operation). Connectors are executed in the order of declaration  
+   **Community**:    Execution duration is not limited. If the connector never finishes, the process instantiation will never finish
```
Comment (rbioteau) [inline md/execution-sequence-states-and-transactions.md]: trim spaces

File: md/execution-sequence-states-and-transactions.md [hunk 2]
```diff
@@ -4,127 +4,137 @@ This page describes the sequence of actions performed when a process instance is
...
-12. The process is instantiated, the API call finish and the Engine executes the process flow asynchronously.
+7. The Engine creates and initializes business data
+1. **Subscription only**: The Engine creates and initializes search indexes
+1. The Engine initializes process instance Documents with default value coming from the definition or with submitted files
+1. Operations (that does not initialize data) are executed. Operations at process instantiation level can either come from V6 forms or from calling process instances (for processes stared by call-activities)
+1. The Engine instantiates and executes the "on enter" connectors (evaluate input expression, execute, evaluate output operation). Connectors are executed in the order of declaration  
+   **Community**:    Execution duration is not limited. If the connector never finishes, the process instantiation will never finish  
+   **Subscription**: Execution duration of a connector is limited to a maximum of 5 minutes by default. This timeout value can be changed by configuration. Above this limit,
+   the connector execution is aborted and the process instantiation is considered failed.
```
Comment (rbioteau) [inline md/execution-sequence-states-and-transactions.md]: remove linebreak

File: md/execution-sequence-states-and-transactions.md [hunk 3]
```diff
@@ -4,127 +4,137 @@ This page describes the sequence of actions performed when a process instance is
...
-For a process step, the sequence of actions is as follows:
+For a user step, the sequence of actions is as follows:
 
-1. The Engine initializes step variables with their default values.
-2. The Engine executes actor filters.
-3. The Engine launches connectors on step enter.
-4. The Engine evaluates the Step dynamic name and description.
-5. They will be evaluated only once.
+1. The Engine initializes task variables with their default values
+1. The Engine executes actor filters and then potentially assigns the task to a user
```
Comment (rbioteau) [inline md/execution-sequence-states-and-transactions.md]: 2.

File: md/execution-sequence-states-and-transactions.md [hunk 4]
```diff
@@ -4,127 +4,137 @@ This page describes the sequence of actions performed when a process instance is
...
+1. The Engine launches the "on enter" connectors
```
Comment (rbioteau) [inline md/execution-sequence-states-and-transactions.md]: 3.

File: md/execution-sequence-states-and-transactions.md [hunk 5]
```diff
@@ -4,127 +4,137 @@ This page describes the sequence of actions performed when a process instance is
...
+1. The Engine evaluates the task dynamic name and description. This will be evaluated only once
```
Comment (rbioteau) [inline md/execution-sequence-states-and-transactions.md]: 4.

File: md/execution-sequence-states-and-transactions.md [hunk 6]
```diff
@@ -4,127 +4,137 @@ This page describes the sequence of actions performed when a process instance is
...
 
 The user task is now in **ready** state and is waiting for a user to submit it.
 
-## Step submission sequence
+## User Step submission sequence
+
+1. The Portal displays the form to the user
+1. The user enters data in the form
```
Comment (rbioteau) [inline md/execution-sequence-states-and-transactions.md:47]: 2.

File: md/execution-sequence-states-and-transactions.md [hunk 7]
```diff
@@ -4,127 +4,137 @@ This page describes the sequence of actions performed when a process instance is
...
+1. The button of the form having the **Submit task** action in the **UI designer** sends a POST request with the form data to start the step
```
Comment (rbioteau) [inline md/execution-sequence-states-and-transactions.md:48]: etc...

File: md/execution-sequence-states-and-transactions.md [hunk 8]
```diff
@@ -4,127 +4,137 @@ This page describes the sequence of actions performed when a process instance is
...
 
-### Execution of a step
+
+## Execution of a step
 
 1. Get the current state of the step.
-2. Execute the state's behavior.
-3. Find the next state of the flow node and set it as the current state.
+1. Execute the state's behavior.
+1. Find the next state of the flow node and set it as the current state.
```
Comment (rbioteau) [inline md/execution-sequence-states-and-transactions.md:72]: 2.

File: md/execution-sequence-states-and-transactions.md [hunk 9]
```diff
@@ -4,127 +4,137 @@ This page describes the sequence of actions performed when a process instance is
...
   1. If the state is stable, the transaction is committed, and the API call is returned.
-  2. If the state is terminal, the transaction is committed, the Engine triggers the asynchronous execution of the followings elements of the process and the API call is returned.
-  3. If the state is neither stable nor terminal, the transition to the next state is scheduled asynchronously.
+  1. If the state is terminal, the transaction is committed, the Engine triggers the asynchronous execution of the followings elements of the process and the API call is returned.
+  1. If the state is neither stable nor terminal, the transition to the next state is scheduled asynchronously.
```
Comment (rbioteau) [inline md/execution-sequence-states-and-transactions.md:75]: 2.

File: md/execution-sequence-states-and-transactions.md [hunk 10]
```diff
@@ -4,127 +4,137 @@ This page describes the sequence of actions performed when a process instance is
...
+
 
 1. The Engine commits the transaction and then submits a work to execute the connectors asynchronously. The connectors are executed outside any transaction and thus are not a problem for the data integrity if the execution takes too long.
-2. As soon as there is a free slot in the WorkService, it executes the work, which is in fact the connector execution.
-3. When a connector execution is finished, if there are other connectors, they are executed in the same way. If there are no other connectors, the Engine continues to execute the state's behavior by triggering a new work.
-4. When the Engine executes a state's behavior, it updates the display name, and then sets the activity to the state "Ready". As this is a stable state, the Engine commits the transaction and stops.
-5. The state "Ready" will then be executed through an API call.
+1. As soon as there is a free slot in the Work Service, it executes the work, which is in fact the connector execution.
+1. When a connector execution is finished, if there are other connectors, they are executed in the same way. If there are no more connectors, the Engine continues to execute the state's behavior by triggering a new work.
```
Comment (rbioteau) [inline md/execution-sequence-states-and-transactions.md:114]: 2.

**Review by rbioteau — CHANGES_REQUESTED** [review]

File: md/execution-sequence-states-and-transactions.md (hunk 3 above)
Comment (educhastenier) [reply md/execution-sequence-states-and-transactions.md]: no need to have proper numbers for numbering to work, and that allows the reordering of the lines without the need for re-number them

File: md/execution-sequence-states-and-transactions.md (hunk 2 above)
Comment (educhastenier) [reply md/execution-sequence-states-and-transactions.md]: The linebreak is here for lisibility when editing the code.
Having lines of 2000 characters is counter-productive.

**Comment by educhastenier** [comment]
retest this please

File: md/execution-sequence-states-and-transactions.md [hunk 11]
```diff
@@ -4,127 +4,137 @@ This page describes the sequence of actions performed when a process instance is
...
-2. The user enters data in the form. 
-3. The button of the form having the **Start process** action in the **UI designer** send a POST request with the form data to start the process. 
-4. The submitted data are validated against the **process instantiation contract**.
-5. If the contract is valid the process instance is created and input values are persisted in database.
+1. The Portal displays the process instantiation form to the user
+1. The user enters data in the form
+1. The button of the form having the **Start process** action in the **UI designer** sends a POST request with the form data to start the process
+1. The submitted data are validated against the **process instantiation contract**
+1. If the contract is valid the process instance is created and input values are persisted in database
 6. Process data instances are created and initialized in the order of declaration using
```
Comment (rbioteau) [inline md/execution-sequence-states-and-transactions.md]: add a `:`

File: md/execution-sequence-states-and-transactions.md [hunk 12]
```diff
@@ -4,127 +4,137 @@ This page describes the sequence of actions performed when a process instance is
...
   * An **operation** declared on the process that saves its value to the process data
```
Comment (rbioteau) [inline md/execution-sequence-states-and-transactions.md]: indent bullet points

**Review by rbioteau — CHANGES_REQUESTED** [review]

**Review by rbioteau — APPROVED** [review]

**Pull request merged**